In [3]:
import pandas as pd
from scipy import stats as stats
from scipy.stats import chi2_contingency
from scipy.stats import chi2

In [5]:
custom= pd.read_csv('/content/Costomer+OrderForm.csv')


In [6]:
custom.head()

Phillippines   Indonesia       Malta       India
0   Error Free  Error Free   Defective  Error Free
1   Error Free  Error Free  Error Free   Defective
2   Error Free   Defective   Defective  Error Free
3   Error Free  Error Free  Error Free  Error Free
4   Error Free  Error Free   Defective  Error Free

In [7]:
print(custom['Phillippines'].value_counts(),custom['Indonesia'].value_counts(),custom['Malta'].value_counts(),custom['India'].value_counts())

Error Free    271
Defective      29
Name: Phillippines, dtype: int64 Error Free    267
Defective      33
Name: Indonesia, dtype: int64 Error Free    269
Defective      31
Name: Malta, dtype: int64 Error Free    280
Defective      20
Name: India, dtype: int64


In [8]:
observed=([[271,267,269,280],[29,33,31,20]])

In [9]:
observed

[[271, 267, 269, 280], [29, 33, 31, 20]]

In [10]:
stat, p, dof, expected = chi2_contingency([[271,267,269,280],[29,33,31,20]])

In [11]:
stat

3.858960685820355

In [12]:
p

0.2771020991233135

In [13]:
print('dof=%d' % dof)
print(expected)

dof=3
[[271.75 271.75 271.75 271.75]
 [ 28.25  28.25  28.25  28.25]]


In [15]:
alpha = 0.05
prob=1-alpha
critical = chi2.ppf(prob, dof)
print('probability=%.3f, critical=%.3f, stat=%.3f' % (prob, critical, stat))
if abs(stat) >= critical:
	print('Dependent (reject H0),variables are related')
else:
	print('Independent (fail to reject H0), variables are not related')

probability=0.950, critical=7.815, stat=3.859
Independent (fail to reject H0), variables are not related


In [16]:
print('significance=%.3f, p=%.3f' % (alpha, p))
if p <= alpha:
	print('Dependent (reject H0)')
else:
	print('Independent (fail to reject H0)')

significance=0.050, p=0.277
Independent (fail to reject H0)


critical=7.815

stat=3.859

probability=0.950

alpha = 0.05

Since Stat value > critical value, we will fail to accept H0 because both variables are independent

Since P value is < alpha value, we will fail to accept H0, that is there is no significant defect between 4 centers.